File: Tarea2/Stage 1/src/main/java/com/example/stage2/VideoFollower.java
package com.example.stage2;

import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.HBox;

public class VideoFollower extends Subscriber {
   public VideoFollower(String name, String topicName) {
      super(name, topicName);
      //??
      view =new HBox ();
      video =new Button("Nothing yet");
      view.getChildren().addAll(new Label(" "+topicName+" -> "+name+": "),video);
      //??
   }
   public void update(String message) {
      //??
      video.setText(message);
      //??
   }
   public HBox getView() {return view;}
   private HBox view;
   private Button video;
}

File: Tarea3/Stage1/videofollower.cpp
#include "videofollower.h"
#include <QPushButton>
#include <QVBoxLayout>

VideoFollower::VideoFollower(const QString& name, const QString& topicName, QWidget* parent)
    : QWidget(parent), Subscriber(name, topicName)
{
    QVBoxLayout* layout = new QVBoxLayout(this);

    button = new QPushButton("Sin URL", this);
    button->setEnabled(false); // desactivado en Etapa 1

    layout->addWidget(button);
}

void VideoFollower::update(const QString& message) {
    button->setText(message);
}
